### PR TITLE
Merge CONTRIBUTING.md into website contributing guide.

### PR DESCRIPTION
**Pull request by ScottTodd**
Following up on some ideas noted at https://github.com/openxla/iree/issues/15116#issuecomment-1791036129.

We had previously split "contributing" docs between the root `CONTRIBUTING.md` and a separate developer doc. Now that the developer doc is published on our website at https://iree.dev/developers/general/contributing/, we can move more of the documentation there and just have the root file point at the website.

Preview for review: https://scotttodd.github.io/iree/developers/general/contributing/

* I rewrote and reordered most of the sections
* I have more ideas for how to streamline these docs... trying to not spent too much time on this right now though 😛 
  * Separate "infrastructure" page explaining how to interact with the CI (build/test/benchmark/etc.)
  * Deemphasize "contributor tips" and/or add more tips (like how to set up Visual Studio Code)

I took some inspiration from https://llvm.org/docs/Contributing.html and https://llvm.org/docs/DeveloperPolicy.html.

### CODE REVIEW

File: docs/website/docs/developers/general/contributing.md [hunk 1]
```diff
@@ -2,21 +2,125 @@
...
+  description should match the PR body (this is the default behavior).
+  Accordingly, please write these as you would a helpful commit message.
+
+It is assumed that the PR author will merge their change unless they ask
+someone else to merge it for them (e.g. because they don't have write access
+yet).
+
+### :octicons-git-merge-16: Obtaining commit access
+
+!!! note "TODO(jpienaar): write this :)"
```
Comment (ScottTodd) [inline docs/website/docs/developers/general/contributing.md]: (I can also write this once you have the form ready to link to)

Should maybe also add a link to that in https://github.com/openxla/community ?

File: docs/website/docs/developers/general/contributing.md [hunk 2]
```diff
@@ -2,21 +2,134 @@
...
+
+### :octicons-code-review-16: Code reviews
+
+All submissions, including submissions by maintainers, require review. We
+use GitHub pull requests (PRs) for this purpose. Consult
+[GitHub Help](https://help.github.com/articles/about-pull-requests/) for more
+information on using pull requests.
+
+* Please keep PRs small (focused on a single issue) to make reviews and later
+  culprit-finding easier.
```
Comment (stellaraccident) [inline docs/website/docs/developers/general/contributing.md:67]: Please add:

`* You may see trusted core contributors bending this rule for project maintenance and major subsystem renovation. If you feel like the rules aren't working for a certain situation, please ask as we bias towards pragmatism for cases that require it.`

**Review by stellaraccident — APPROVED** [review]
One statement I would like added.

**Review by stellaraccident — APPROVED** [review]

File: docs/website/docs/developers/general/contributing.md [hunk 3]
```diff
@@ -2,21 +2,160 @@
...
+
+* The squashed commit summary should match the PR title and the commit
+  description should match the PR body (this is the default behavior).
+  Accordingly, please write these as you would a helpful commit message.
+
+It is assumed that the PR author will merge their change unless they ask
+someone else to merge it for them (e.g. because they don't have write access
+yet).
+
+### :octicons-git-merge-16: Obtaining commit access
```
Comment (ScottTodd) [inline docs/website/docs/developers/general/contributing.md:108]: Wrote this section :)

Live preview: https://scotttodd.github.io/iree/developers/general/contributing/#obtaining-commit-access

image

The form goes to Jacques to review access requests

**Pull request merged**